components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,200 @@
+            Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        } else {
+            log.warn("Provided template type does not exist");
+            Utils.setJsonPayLoad(axis2MessageContext,
+                    Utils.createJSONErrorObject("Received type " + templateType + " is not a valid template type"));
+        }
+    }
+
+    /**
+     * Sets the name of the template to the JSON Array
+     *
+     * @param jsonList body of the json object
+     * @param name     template name
+     */
+    private void addToJsonList(JSONArray jsonList, String name) {
+        //Create new JSONObject for the template resource
+        JSONObject templateObject = new JSONObject();
+        templateObject.put(Constants.NAME, name);
+        //Add the template JSONObject into the JSON array on the parent object
+        jsonList.put(templateObject);
+    }
+
+    /**
+     * Set the information related to a specific template to the Axis2MessageContext as a JSON object
+     *
+     * @param templateName   name of the template
+     * @param templateType   type of the template
+     */
+    private void populateTemplateData(MessageContext messageContext, String templateName, String templateType) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        if (ENDPOINT_TEMPLATE_TYPE.equalsIgnoreCase(templateType)) {
+            Template endpointTemplate = synapseConfiguration.getEndpointTemplate(templateName);
+            Utils.setJsonPayLoad(axis2MessageContext, getEndpointTemplateAsJson(endpointTemplate));
+        } else if (SEQUENCE_TEMPLATE_TYPE.equalsIgnoreCase(templateType)) {
+            TemplateMediator sequenceTemplate = synapseConfiguration.getSequenceTemplate(templateName);
+            Utils.setJsonPayLoad(axis2MessageContext, getSequenceTemplateAsJson(sequenceTemplate));
+        }
+    }
+
+    /**
+     * Returns the JSON representation of a endpoint template
+     *
+     * @param endpointTemplate
+     */
+    private JSONObject getEndpointTemplateAsJson(Template endpointTemplate) {
+        JSONObject endpointTemplateJSONObject = new JSONObject();
+        endpointTemplateJSONObject.put(Constants.NAME, endpointTemplate.getName());
+        endpointTemplateJSONObject.put("Parameters", endpointTemplate.getParameters());
+
+        return endpointTemplateJSONObject;
+    }
+
+    /**
+     * Returns the JSON representation of a sequence template
+     *
+     * @param sequenceTemplate
+     */
+    private JSONObject getSequenceTemplateAsJson(TemplateMediator sequenceTemplate) {
+        JSONObject sequenceTemplateJSONObject = new JSONObject();
+        sequenceTemplateJSONObject.put(Constants.NAME, sequenceTemplate.getName());
+        sequenceTemplateJSONObject.put("Parameters", sequenceTemplate.getParameters());
```

Comment:
Shall we define a constant for "Parameters"

Reply:
+1

Reply:
done